### PR TITLE
[stable/kong] introduce support for dbless Ingress Controller

### PR DESCRIPTION
**Pull request by hbagdi**
Signed-off-by: Harry Bagdi <harrybagdi@gmail.com>

#### What this PR does / why we need it:
- introduces support to deploy Kong as an Ingress Controller without a database

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] title of the PR contains starts with chart name e.g. `[stable/chart]`

### CODE REVIEW

**Comment by hbagdi** [comment]
The CI failed but doesn't even have any logs.

@unguiculus  Any clue?

**Comment by hbagdi** [comment]
ping @davidkarlsen as well

**Comment by hbagdi** [comment]
In an attempt to get logs for the failure, I'll rerun the test.
/test pull-charts-e2e

**Comment by hbagdi** [comment]
ping @scottrigby for any insight.

File: stable/kong/README.md [hunk 1]
```diff
@@ -186,6 +186,10 @@ To deploy the ingress controller together with
 kong run the following command:
 
 ```bash
+# with a database
```
Comment (shashiranjan84) [inline stable/kong/README.md]: Is this comment correct? I think you mean `without a database`

File: stable/kong/README.md [hunk 2]
```diff
@@ -186,6 +186,10 @@ To deploy the ingress controller together with
...
+helm install stable/kong --set ingressController.enabled=true \
+  --set postgresql.enabled=false --set env.database=off
+# without a database
```
Comment (shashiranjan84) [inline stable/kong/README.md]: and this should be `with a database`

File: stable/kong/requirements.lock [hunk 3]
```diff
@@ -1,9 +1,9 @@
 dependencies:
 - name: postgresql
   repository: https://kubernetes-charts.storage.googleapis.com/
-  version: 3.18.3
+  version: 3.9.5
```
Comment (shashiranjan84) [inline stable/kong/requirements.lock:4]: Not matching with https://github.com/helm/charts/pull/13475/files#diff-509c05d7eb2ab1aaa0709017c4b59a61R3

Comment (hbagdi) [reply stable/kong/requirements.lock:4]: This should be fine since we are pinning 3.9 and any patch version bumps are acceptable.

**Review by shashiranjan84 — APPROVED** [review]
/lgtm

**Pull request merged**